### PR TITLE
Move channels and their mutex to a separate class

### PR DESCRIPTION
**Pull request by KacperKluka**
I've tried to implement the request from [this comment](https://github.com/ably/ably-asset-tracking-android/pull/773#pullrequestreview-1135798681) from https://github.com/ably/ably-asset-tracking-android/pull/773 and wanted to create a quick draft PR to validate whether I'm following the correct approach.

### CODE REVIEW

File: common/src/main/java/com/ably/tracking/common/AblyChannelsGuard.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    private val channelsMutex = Mutex()
+
+    fun getChannel(trackableId: String): Channel? = channels[trackableId]
+
+    fun getChannelTrackableIds(): Set<String> = channels.keys
+
+    fun isChannelCreated(trackableId: String): Boolean = channels.containsKey(trackableId)
+
+    suspend fun startConnectingChannel(trackableId: String, onConnectingFirstChannel: suspend () -> Unit) {
```
Comment (davyskiba) [inline common/src/main/java/com/ably/tracking/common/AblyChannelsGuard.kt:29]: This name confused me a bit, I was expecting to see a channel connection inside this. Maybe we could rename it to something like `addConnectingChannel`?

File: common/src/main/java/com/ably/tracking/common/Ably.kt [hunk 2]
```diff
@@ -413,20 +384,22 @@ constructor(
     }
 
     override suspend fun disconnect(trackableId: String, presenceData: PresenceData): Result<Unit> {
-        return if (channels.contains(trackableId)) {
-            val channelToRemove = channels[trackableId]!!
+        return if (channelsGuard.isChannelCreated(trackableId)) {
+            val channelToRemove = channelsGuard.getChannel(trackableId)!!
```
Comment (davyskiba) [inline common/src/main/java/com/ably/tracking/common/Ably.kt:388]: This force unwrap could be avoided with something like
`  val channelToRemove = channelsGuard.getChannel(trackableId) ?: return Result.success(Unit)`

**Comment by davyskiba** [comment]
I like this approach :)

**Comment by KacperKluka** [comment]
Closing as we won't use the mutex based approach

**Pull request closed without merging**